src/foam/nanos/auth/CapabilityAuthService.js

```diff
@@ -59,16 +61,32 @@ foam.CLASS({
         Capability cap = (Capability) capabilityDAO.find(permission);
         if ( cap != null && cap.isDeprecated(x) ) return getDelegate().checkUser(x, user, permission);
 
+        Date now = new Date();
+        Predicate capabilityScope = AND(
+          EQ(UserCapabilityJunction.SOURCE_ID, user.getId()),
+          OR(
+            NOT(HAS(UserCapabilityJunction.EXPIRY)),
+            AND(
+              HAS(UserCapabilityJunction.EXPIRY),
+              GTE(UserCapabilityJunction.EXPIRY, now)
+            )
+          )
+        );
+
         DAO userCapabilityJunctionDAO = (DAO) x.get("userCapabilityJunctionDAO");
 
         if ( userCapabilityJunctionDAO.find(
           AND(
-            EQ(UserCapabilityJunction.SOURCE_ID, user.getId()),
+            capabilityScope,
             EQ(UserCapabilityJunction.TARGET_ID, permission),
             EQ(UserCapabilityJunction.STATUS, CapabilityJunctionStatus.GRANTED)
           )) != null ) return true;
         
-        List<UserCapabilityJunction> userCapabilityJunctions = ((ArraySink) user.getCapabilities(x).getJunctionDAO().where(EQ(UserCapabilityJunction.SOURCE_ID, user.getId())).select(new ArraySink())).getArray();
+        List<UserCapabilityJunction> userCapabilityJunctions = ((ArraySink) user.getCapabilities(x).getJunctionDAO()
+          .where(
+            capabilityScope
```

Comment:
This doesn't need to be on its own line.

Reply:
Updated